cubit_shopper/lib/models/cart.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_bloc/flutter_bloc.dart';
+import '../models/catalog.dart';
+
+class CartModelState {
+  CartModelState({required this.items, required this.totalPrice});
+  final List<Item> items;
+  final int totalPrice;
+}
+
+class CartModel extends Cubit<CartModelState> {
+  CartModel() : super(CartModelState(items: [], totalPrice: 0));
+
+  /// The private field backing [catalog].
+  late CatalogModel _catalog;
+
+  /// Internal, private state of the cart. Stores the ids of each item.
+  final List<int> _itemIds = [];
+
+  /// The current catalog. Used to construct items from numeric ids.
+  CatalogModel get catalog => _catalog;
+
+  void _emitUpdatedState() {
+    final items = _itemIds.map((id) => _catalog.getById(id)).toList();
+    final totalPrice = items.fold(0, (total, current) => total + current.price);
+    emit(CartModelState(items: items, totalPrice: totalPrice));
+  }
+
+  set catalog(CatalogModel newCatalog) {
+    _catalog = newCatalog;
+    // Notify listeners, in case the new catalog provides information
+    // different from the previous one. For example, availability of an item
+    // might have changed.
+    _emitUpdatedState();
+  }
+
+  /// Adds [item] to cart. This is the only way to modify the cart from outside.
+  void add(Item item) {
+    _itemIds.add(item.id);
+    // This line tells [Model] that it should rebuild the widgets that
```

Comment:
Not quite sure on the history of this comment, but it is no longer true in the current context.